fe1-web/network/NetworkConnection.ts

```diff
@@ -61,10 +53,28 @@ export class NetworkConnection {
           throw new NetworkError(`Received a response which id = ${response.id} does not match any pending requests`);
         }
 
+        // a response was received, we clear the timeout and the pending query from the pending map
+        clearTimeout(pendingResponse.timeoutId);
         this.payloadPending.delete(response.id);
 
-        pendingResponse.resolvePromise(response);
-        this.onMessageHandler(response);
+        // reject the promise if the response is negative
+        if (response.isPositiveResponse()) {
+          pendingResponse.resolvePromise(response);
+          try {
+            this.onMessageHandler(response);
+          } catch (e) {
+            console.error('Exception encountered when giving response : ', response, ' to its handler.\n', e);
+          }
+        } else {
+          // Note : impossible to have an undefined error at this stage due to isPositiveResponse()
+          pendingResponse.rejectPromise(
```

Comment:
we could consider returning a specific kind of error here, which could be constructed with the error code + description (i.e. as 2 separate fields).
This way, it'll be easier to process at a higher level.
E.g. in the GUI "Could not execute operation: ${response.error.description}"

Reply:
Good idea! See commit 4f173068